### ISSUE TITLE
Gang: Flash Mobbing

### ISSUE THREAD

**Issue by LJNeon** [bitburner-official/bitburner-src#1066]
Flash mobbing is when gang members are only set to territory warfare for the territory & power update and are set to other tasks for every other gang update. They still get the full benefit of the territory warfare task.

Currently the suggested solutions are:
- Simplify this behavior by switching to a checkbox for gang members to participate in territory warfare, unrelated to the task they are working on.
- Randomize which gang update is the territory & power update, still only happening once per 100 cycles (20 seconds).
- Increase territory gains on clash win based on how many cycles gang members have spent on territory warfare since the last territory & power update.
- Increase power gains from the territory warfare task based on how many cycles gang members have spent on the task since the last territory & power update.
- Do nothing, the current behavior is fine.

**Comment by Semanual** [comment]
About the first one: why would anyone want to uncheck participating in territory warfare, if it was only a checkbox? If this get selected as the new behavior, it has to create something else as a demerit for enabling too soon, else it'll just be forever checked.

**Comment by LJNeon** [comment]
Maybe a small efficiency decrease to the selected task is applied when the territory warfare box is checked? That seems like the obvious answer and would still align with current behavior (have to spend a gang update on territory warfare instead of their designated task).

**Comment by Semanual** [comment]
I feel like this is just simplifying flash mobbing, you still have it but you masked it behind another UI (and API calls), and made it easier to script. To align with the current behavior of bladeburners, I think the third option is the best, if you are going to remove flash mobbing.
But in my opinion, flash mobbing isn't broken, and actually feels like the natural gang progression, as if gangs were actually designed to be this way. You grow a little behind the other gangs, until finally beating them. If you had to first build up stats, all the gangs become so much more powerful than you that bit_fluming is just easier, so you essentially render gangs useless.

**Comment by Semanual** [comment]
By the way, theoretically NPC gangs are also flashmobbing, they earn power every warfare tick and grow their own power every tick too, so if you think it's a problem, I think the AI's behavior should be reworked too.

**Comment by LJNeon** [comment]
Yea it is just simplifying it, that was the goal Snarling had when suggesting the first one. And I disagree about it being a natural progression, and that's not at all what your reasoning seems to suggest. It seems like you're just saying we need to buff gangs in some way if we remove it because it's currently necessary for gangs to succeed (although I think you're just used to it, it's very possible to run gangs without ever flash mobbing).

But currently I will say that I support either just the third option or both the second and third option. It seems better to make flash mobbing unnecessary by providing a superior alternative.

**Comment by LJNeon** [comment]
But currently I will say that I support either just the third option or both the second and third option. It seems better to make flash mobbing unnecessary by providing a better alternative.

**Comment by LJNeon** [comment]
> By the way, theoretically NPC gangs are also flashmobbing, they earn power every warfare tick and grow their own power every tick too, so if you think it's a problem, I think the AI's behavior should be reworked too.

I mean I did spot a comment about adding police clashes to gangs when I double-checked exactly how gang internals work.

**Comment by Semanual** [comment]
> it's very possible to run gangs without ever flash mobbing

True, but it will take so long to grow up them that you will probably be in the verge of destroying WD by the time you can.
> I mean I did spot a comment about adding police clashes to gangs when I double-checked exactly how gang internals work.

As if police were an eighth faction that does not earn rep? (and yeah, honestly, I miss the police in this game, you can just do anything without ever being caught)

**Comment by LJNeon** [comment]
> True, but it will take so long to grow up them that you will probably be in the verge of destroying WD by the time you can.

I think that's an exaggeration, it's slower but not so slow that it makes gangs worthless.

> As if police were an eighth faction that does not earn rep? (and yeah, honestly, I miss the police in this game, you can just do anything without ever being caught)

The [comment](https://github.com/bitburner-official/bitburner-src/blob/dev/src/Gang/Gang.ts#L1-L5) isn't exactly detailed, so we could probably do a lot with police (like using them to slow down the AI gangs).

**Comment by Semanual** [comment]
And while I agree that intuitively changing to territory warfare in the last second and getting the full power is strange, I think that maybe some rewriting could solve that, as if it was like you being open to be invited for warfare in those ticks, instead of actively doing territory warfare the entire time.

**Comment by Semanual** [comment]
> The [comment](https://github.com/bitburner-official/bitburner-src/blob/dev/src/Gang/Gang.ts#L1-L5) isn't exactly detailed, so we could probably do a lot with police (like using them to slow down the AI gangs).

Oh right, it says about making it possible for gangs not running out of control... if it gets implemented, I think flash mobbing won't be needed anymore, as the main reason I'm against it is because the more time passes, the harder it is to actually beat other gangs.

**Comment by LJNeon** [comment]
> And while I agree that intuitively changing to territory warfare in the last second and getting the full power is strange, I think that maybe some rewriting could solve that, as if it was like you being open to be invited for warfare in those ticks, instead of actively doing territory warfare the entire time.

I think caldwell's suggestion (about increasing territory gains from a win based on how long the gang members have spent on the task) is more intuitive than that. It makes complete sense to me that devoting more time to the territory warfare task would result in greater territory gains.

**Comment by Semanual** [comment]
Really it makes sense, but TW actually just increase power, right? So, how can it increase territory gains?

**Comment by LJNeon** [comment]
> Really it makes sense, but TW actually just increase power, right? So, how can it increase territory gains?

Sure [looks](https://github.com/bitburner-official/bitburner-src/blob/dev/src/Gang/Gang.ts#L234-L235) like it increases/decreases territory to me.

**Comment by Semanual** [comment]
> > Really it makes sense, but TW actually just increase power, right? So, how can it increase territory gains?
> 
> Sure [looks](https://github.com/bitburner-official/bitburner-src/blob/dev/src/Gang/Gang.ts#L234-L235) like it increases/decreases territory to me.

But those are clashes, not territory warfare

**Comment by LJNeon** [comment]
Ah oops. So the suggestion was probably that territory gain from clashes could be increased based on how many cycles your gang members have been on territory warfare for since the last territory & power update.

**Comment by Jman420** [comment]
I think another option would be a variation on option 3.  With the proposed option 3 flash mobbing would still be useful pre-warfare in order to quickly boost power until the clash chance to win hits a threshold the player is comfortable with (ie. my scripts use 80% before engaging warfare) and warfare gets enabled.

Instead of weighting the territory gains it may make more sense to weight the power gains.  Leave the TW tick as the last tick in the cycle, but weight the power gained by how long the member has been in TW for that cycle.  That would impact both pre & post warfare since power gained during pre-warfare would be reduced for members who split their tasks each cycle and power would reduce faster during warfare since less is being accumulated for each member that splits their tasks.

Players can still micromanage their tasks per tick in a cycle to try to squeeze more efficiency out of their gang.  In pre-warfare a 50/50 split for training & TW might be effective, but once warfare is engaged a 25/75 split may be more effective... or maybe even higher depending on the power levels of the other gangs.

**Comment by LJNeon** [comment]
I really like that idea, I edited it in.

**Comment by User670** [comment]
A dumb idea: the power update tick immediately after assigning this task does nothing, only starting from the tick after that (which guarantees a full tick's length has passed) will it start to take effect.

(yes this *does* mean that the first half-tick after swapping task will be wasted.)
